src/Microsoft.IdentityModel.JsonWebTokens/Json/JsonClaimSet.cs

```diff
@@ -43,16 +43,16 @@ internal IList<Claim> Claims(string issuer)
             return _claims;
         }
 
-        internal IList<Claim> CreateClaims(string issuer)
```

Comment:
Some of the advantages were noted in the PR description.

For example, if you have:
```C#
IList<T> list = ...;
foreach (T item in list) { ... }
```
that allocates an `IEnumerator<T>` object. If you instead have:
```C#
List<T> list = ...;
foreach (T item in list) { ... }
```
it doesn't, instead using the `List<T>.Enumerator` struct for enumeration.

But even just simple accesses, e.g.
```C#
IList<T> list = ...;
return list[2];
```
That involves interface dispatch, which is not only slower than a regular method invocation:
```C#
List<T> list = ...;
return list[2];
```
but without a technology like dynamic PGO, it can't be inlined.